packages/combo-box/src/vaadin-combo-box-data-provider-mixin.js

```diff
@@ -244,6 +238,8 @@ export const ComboBoxDataProviderMixin = (superClass) =>
         this.pageSize = oldPageSize;
         throw new Error('`pageSize` value must be an integer > 0');
       }
+
+      this.__dataProviderController.setPageSize(pageSize);
```

Comment:
nit: looks like all the tests pass with `clearCache()` line below being removed.
But this also happens on `main` so not related to this PR, can be ignored.

Reply:
Yeah, I think we could improve this separately. To be safe, I tested it manually and it works in both WC and Flow.